lib/bemxjst/match.js

```diff
@@ -195,7 +192,7 @@ Match.prototype.exec = function exec(context) {
       }
 
       // All predicates matched!
-      if (j === -1)
+      if (j === template.predicates.length)
```

Comment:
New IE will do it quite fine, old IE are kind of slow anyway? Do we want to have crappy code just because of one old browser?